jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/java/JavaSamProjectWizard.kt

```diff
@@ -77,69 +66,27 @@ abstract class JavaSamProjectTemplate : SamProjectTemplate() {
     }
 }
 
-abstract class JavaGradleSamProjectTemplate : JavaSamProjectTemplate() {
-    override fun postCreationAction(
-        settings: SamNewProjectSettings,
-        contentRoot: VirtualFile,
-        rootModel: ModifiableRootModel,
-        sourceCreatingProject: Project,
-        indicator: ProgressIndicator
-    ) {
-        super.postCreationAction(settings, contentRoot, rootModel, sourceCreatingProject, indicator)
-        val buildFile = locateBuildFile(contentRoot, "build.gradle") ?: return
-
-        val gradleProjectSettings = GradleProjectSettings().apply {
-            withQualifiedModuleNames()
-            externalProjectPath = buildFile.path
-        }
-
-        val externalSystemSettings = ExternalSystemApiUtil.getSettings(rootModel.project, GradleConstants.SYSTEM_ID)
-        externalSystemSettings.setLinkedProjectsSettings(setOf(gradleProjectSettings))
-
-        val importSpecBuilder = ImportSpecBuilder(rootModel.project, GradleConstants.SYSTEM_ID)
-            .forceWhenUptodate()
-            // replaces useDefaultCallback which is removed in 2020.3
-            // FIX_WHEN_MIN_IS_202 remove this callback (since we don't use it)
-            .callback(null)
-            .use(ProgressExecutionMode.IN_BACKGROUND_ASYNC)
-
-        ExternalSystemUtil.refreshProjects(importSpecBuilder)
-    }
-}
-
-class SamHelloWorldMaven : JavaSamProjectTemplate() {
-    override fun getName() = message("sam.init.template.hello_world_maven.name")
-
-    override fun getDescription() = message("sam.init.template.hello_world.description")
-
-    override fun templateParameters(): TemplateParameters = AppBasedTemplate("hello-world", "maven")
-
+abstract class JavaMavenSamProjectTemplate : JavaSamProjectTemplate() {
     override fun postCreationAction(
         settings: SamNewProjectSettings,
         contentRoot: VirtualFile,
         rootModel: ModifiableRootModel,
-        sourceCreatingProject: Project,
         indicator: ProgressIndicator
     ) {
-        super.postCreationAction(settings, contentRoot, rootModel, sourceCreatingProject, indicator)
+        super.postCreationAction(settings, contentRoot, rootModel, indicator)
         val pomFile = locateBuildFile(contentRoot, "pom.xml") ?: return
         val projectsManager = MavenProjectsManager.getInstance(rootModel.project)
-        projectsManager.addManagedFilesOrUnignore(listOf(pomFile))
+        runInEdt {
+            projectsManager.addManagedFilesOrUnignore(listOf(pomFile))
```

Comment:
How was this working not in a `runInEdt`/why is it required to be put in a `runInEdt`? Probably want to add as a comment as well

Reply:
We used to run everything in runInEdit, see the diff for `generateProject`